src/Microsoft.Framework.Configuration/ConfigurationBase.cs

```diff
@@ -21,13 +21,15 @@ public ConfigurationBase(IList<IConfigurationSource> sources)
             _sources = sources;
         }
 
+        public abstract string Path { get; }
+
         public string this[string key]
         {
             get
             {
                 if (string.IsNullOrEmpty(key))
                 {
-                    throw new InvalidOperationException(Resources.Error_EmptyKey);
+                    throw new ArgumentException(Resources.Error_EmptyKey);
```

Comment:
The logic of the indexer is now redundant. Replace the body of the getter with `return GetSection(key).Value;` and the setter with `GetSection(key).Value = value;` and then get rid of `GetPrefix()` for good :smile: